relayer/pkg/chainlink/txm/txm.go

```diff
@@ -125,7 +129,52 @@ func (txm *starktxm) broadcastLoop() {
 	}
 }
 
+func (txm *starktxm) handleNonceErr(ctx context.Context, accountAddress *felt.Felt) error {
+
+	txm.lggr.Debugw("Handling Nonce Validation Error By Resubmitting Txs...", "account", accountAddress)
+
+	// resync nonce so that new queued txs can be unblocked
+	client, err := txm.client.Get()
+	if err != nil {
+		return err
+	}
+
+	chainId, err := client.Provider.ChainID(ctx)
+	if err != nil {
+		return err
+	}
+
+	txm.nonce.Sync(ctx, accountAddress, chainId, client)
+
+	// todo: in the future, revisit resetting txm.txStore.currentNonce
+
+	unconfirmedTxs, err := txm.txStore.GetUnconfirmedSorted(accountAddress)
+	if err != nil {
+		return err
+	}
+
+	// delete all unconfirmed txs and resubmit them to the txm queue
```

Comment:
i had similar thoughts, but then realized that you could have tx4 and tx5 both having nonce issues but you process tx5 first. So potentially all txs need to be retried. 

If there were another tx3 in the queue which was completly fine and was accepted, it's ok to retry that one as well because it'll just fail on retry